email-address-parser/email-addresses.mjs

```diff
@@ -0,0 +1,1087 @@
+
+// email-addresses.js - RFC 5322 email address parser
+// v 5.0.0
+//
+// http://tools.ietf.org/html/rfc5322
+//
+// This library does not validate email addresses.
+// emailAddresses attempts to parse addresses using the (fairly liberal)
+// grammar specified in RFC 5322.
+//
+// email-addresses returns {
+//     ast: <an abstract syntax tree based on rfc5322>,
+//     addresses: [{
+//            node: <node in ast for this address>,
+//            name: <display-name>,
+//            address: <addr-spec>,
+//            local: <local-part>,
+//            domain: <domain>
+//         }, ...]
+// }
+//
+// emailAddresses.parseOneAddress and emailAddresses.parseAddressList
+// work as you might expect. Try it out.
+//
+// Many thanks to Dominic Sayers and his documentation on the is_email function,
+// http://code.google.com/p/isemail/ , which helped greatly in writing this parser.
+
+export function parse5322(opts) {
```

Comment:
Meanwhile I understood that this was a local modification on top of the original version. But we can do better, we can go upstream first (provided upstream is still merging): https://github.com/jackbearheart/email-addresses/pull/68